File: Week6/ProblemSet/sentimental-mario-more/mario.py
import cs50


def main():
    height = 0

    while height < 1 or height > 8:
        height = cs50.get_int("Height: ")

    print(build(height))


def build(h):
    pyramid = ""

    for i in range(1, h + 1):
        spaces = " " * (h - i)
        hashes = "#" * i
        pyramid += f"{spaces}{hashes}  {hashes}\n"

    if pyramid:
        pyramid = pyramid.rstrip("\n")

    return pyramid


if __name__ == "__main__":
    main()

File: CS50xIntroToComputerScience/Week1/ProblemSet/mario/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void build(int h);
void get_height(int *height);

int main(void)
{
    int height;
    get_height(&height);
    build(height);
}

void build(int h)
{
    for (int i = 1; i <= h; i++)
    {
        // Print spaces
        for (int j = 0; j < h - i; j++)
        {
            printf(" ");
        }

        // Print hashes
        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

void get_height(int *height)
{
    do
    {
        *height = get_int("Height: ");
    }
    while (*height < 1 || *height > 8);
}